pkg/metrics/builder.go

```diff
@@ -67,7 +81,7 @@ func (b *Builder) CreateMetricsFactory(namespace string) (metrics.Factory, error
 		b.handler = promhttp.Handler()
 		return metricsFactory, nil
 	}
-	if b.Backend == "expvar" {
+	if b.Backend == defaultMetricsBackend {
```

Comment:
this function will do dispatching based on the requested backend, so it needs to check for the exact backend name like "expvar" or "prometheus". The value of `defaultMetricsBackend` constant can change independently (ideally I'd want prometheus to be the default, once we have support for it).